`CMake 4.0` dropped support for `CMake <3.5`

### Expected behavior

Successful compilation with `CMake 4.0`

### Actual behavior

`CMake` removed with their newest release (`4.0.0`) compatibility with `CMake < 3.5`.
It can be overriden, but best it would be addressed upstream. Especially if the compatibility was/is necessary.

### Steps to reproduce

Compile the source with CMake 4.0

```
CMake Error at CMakeLists.txt:5 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
```

### Environment

**Operating system:**

Linux

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,7 +25,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.10)
 
 project(openvas-smb
   VERSION 22.5.7

